SYCL/ESIMD/api/functional/ctors/ctor_default.cpp

```diff
@@ -84,52 +82,53 @@ struct const_ref {
   }
 };
 
-// Functor with the main test routine to iterate over the pre-determined
-// datatypes
-template <typename DataT, int NumElems, typename TestCaseT> struct test {
-  bool operator()(sycl::queue &queue, const std::string &data_type) {
-    bool passed{true};
-
+// Struct that calls simd in provided context and then verifies obtained result.
+struct Verifier {
+  template <typename DataT, int NumElems, typename TestCaseT>
+  static bool verify(sycl::queue &queue, const std::vector<DataT> ref_data,
+                     const std::string &data_type) {
+    bool passed = true;
     DataT default_val{};
-    using AllocatorT = sycl::usm_allocator<DataT, sycl::usm::alloc::shared>;
 
-    std::vector<DataT, AllocatorT> result{NumElems, AllocatorT{queue}};
+    shared_vector<DataT> result(NumElems, shared_allocator<DataT>{queue});
+
     queue.submit([&](sycl::handler &cgh) {
       auto out = result.data();
       cgh.single_task<Kernel<DataT, NumElems, TestCaseT>>(
           [=]() SYCL_ESIMD_KERNEL {
             TestCaseT::template call_simd_ctor<DataT, NumElems>(out);
           });
     });
-    for (const auto &it : result) {
-      if (it != default_val) {
+
+    for (size_t i = 0; i < result.size(); ++i) {
+      if (result[i] != default_val) {
```

Comment:
Once again, do we need to output index? The only concern I have is for the `simd<T,1>` failures output though.

Reply:
I think that all is ok, for example output in case `NumElements == 1`:
```
Failed for initializer failed, retrieved: 0, expected: 0, at index: 0, with data type: char, with num elements: 1
Failed for initializer failed, retrieved: 0, expected: 0, at index: 0, with data type: unsigned char, with num elements: 1
Failed for initializer failed, retrieved: 0, expected: 0, at index: 0, with data type: signed char, with num elements: 1
Failed for initializer failed, retrieved: 0, expected: 0, at index: 0, with data type: short, with num elements: 1
Failed for initializer failed, retrieved: 0, expected: 0, at index: 0, with data type: unsigned short, with num elements: 1
Failed for initializer failed, retrieved: 0, expected: 0, at index: 0, with data type: int, with num elements: 1
Failed for initializer failed, retrieved: 0, expected: 0, at index: 0, with data type: unsigned int, with num elements: 1
```